### PR TITLE
Update nginx to use our own image

### PR DESCRIPTION
**Pull request by Fraggle**
# Description

This PR updates the nginx configuration to use our own custom image.

Changes include:
- Adding a new `.gcloudignore-nginx` file
- Creating a new GitHub Actions workflow for deploying nginx (`deploy-nginx.yml`)
- Modifying the proxy configuration and deployment in Kubernetes
- Adding a new `nginx.Dockerfile` for building our custom nginx image

# Risk

Already live (ran the commands locally)

# Deploy

Deploy nginx, apply infra.

### CODE REVIEW

File: k8s/dust-kube-private/configmaps/proxy-configmap.yaml [hunk 1]
```diff
@@ -40,9 +40,3 @@ data:
       }
     }
 
-    stream {
```
Comment (Fraggle) [inline k8s/dust-kube-private/configmaps/proxy-configmap.yaml:43]: this was a tentative for socks, removing it in favor of dante.

**Comment by fontanierh** [comment]
Just to make sure i fully get it. Why do we need our own image?

**Comment by Fraggle** [comment]
@fontanierh the official images do not include proxy connect extension so we were relying on someone’s 5yo images. It was using a quite old nginx. It was not mandatory to use our own image but it allowed me to use the latest nginx.

File: nginx.Dockerfile [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+    && runDeps="$(scanelf --needed --nobanner --format '%n#p' /usr/sbin/nginx /tmp/envsubst | tr ',' '\n' | sort -u | awk 'system("[ -e /usr/local/lib/" $1 " ]") == 0 { next } { print "so:" $1 }')" \
+    && apk add --no-cache --virtual .nginx-rundeps $runDeps \
+    && apk del .build-deps \
+    && apk del .gettext \
+    && mv /tmp/envsubst /usr/local/bin/ \
+    && apk add --no-cache tzdata \
+    && ln -sf /dev/stdout /var/log/nginx/access.log \
+    && ln -sf /dev/stderr /var/log/nginx/error.log
+
+RUN nginx -V 2>&1
```
Comment (fontanierh) [inline nginx.Dockerfile:84]: ooc is it debug ?

Comment (fontanierh) [inline nginx.Dockerfile:84]: ooc is it debug ?

**Review by fontanierh — APPROVED** [review]
LGTM

File: nginx.Dockerfile (hunk 2 above)
Comment (Fraggle) [reply nginx.Dockerfile:84]: I believe, It's to show the version at the start of the logs (i copy pasted from the other image to be honest).

**Pull request merged**